### PR TITLE
mmligner

### PR DESCRIPTION
**Pull request by Koesed96**
## Description
Provides a Wrapper for the MMligner.

## Todos
  - [ ] expand tests
  - [x] add tutorial
## Status
- [ ] Ready to go

### CODE REVIEW

File: structuralalignment/superposition/mmligner.py [hunk 1]
```diff
@@ -103,18 +104,17 @@ def _parse(self, output):
...
 
             elif line.startswith(b"I(A & <S,T>)"):
                 ivalue = float(line.split()[4])
 
         alignment = fasta.FastaFile()
 
         return {
-            "superposed": None,
-            "scores": {"rmsd": rmsd, "score": ivalue},
+            "superposed": atomium.data.File("p_superposed__1.pdb").model,
```
Comment (jaimergp) [inline structuralalignment/superposition/mmligner.py:116]: You can use `atomium.open(..path..).model`

Comment (Koesed96) [reply structuralalignment/superposition/mmligner.py:116]: Oh okay, thanks for the fast reply!

File: docs/tutorials/notebook_mmligner.ipynb [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+    "To get the structures directly from the RCSB, the syntax looks like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "structuralalignment --method=mmligner 2GZ9 5R8T"
```
Comment (jaimergp) [inline docs/tutorials/notebook_mmligner.ipynb:100]: Use IPython magic commands to run in shell:

```suggestion
    "!structuralalignment --method=mmligner 2GZ9 5R8T"
```

File: docs/tutorials/notebook_mmligner.ipynb [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+    "When you want to use structures which are locally saved, do this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "structuralalignment --method=mmligner PATH_OF_2GZ9 PATH_OF_5R8T"
```
Comment (jaimergp) [inline docs/tutorials/notebook_mmligner.ipynb:116]: Same: 

```suggestion
    "!structuralalignment --method=mmligner PATH_OF_2GZ9 PATH_OF_5R8T"
```

File: docs/tutorials/notebook_mmligner.ipynb [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+    "Using MMligner in Python looks like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mmligner = MMlignerAligner()\n",
```
Comment (jaimergp) [inline docs/tutorials/notebook_mmligner.ipynb:176]: You are missing the import!

```suggestion
    "from structuralalignment.superposition.mmligner import MMlignerAligner\n",
    "mmligner = MMlignerAligner()\n",
```

**Comment by Koesed96** [comment]
Sorry for the failing tests. As far as I can tell the problem is that the patches to mmligner are not adopted.

**Comment by jaimergp** [comment]
No probs, I'll take that! Thanks for the changes!

**Comment by hovo1990** [comment]
Hi there, 
When I am trying to use MMLigner, I get the following error:


`
mmligner = MMLignerAligner()

results_mm = mmligner.calculate([mol1_protein, mol2_protein])`

> Current MMLigner wrappers produces accurate RMSD values but slightly shifted structures!
/home/i-am-curious/.conda/envs/pyscience/lib/python3.7/site-packages/MDAnalysis/coordinates/PDB.py:1028: UserWarning: Found no information for attr: 'tempfactors' Using default value of '0.0'
  "".format(attrname, default))
/home/i-am-curious/.conda/envs/pyscience/lib/python3.7/site-packages/MDAnalysis/core/topologyattrs.py:2011: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
  np.array(sorted(unique_bonds)), 4) -------------------------------------------------------------------------- 
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-86-9df85d2189fa> in <module>
1 mmligner = MMLignerAligner()
----> 2 results_mm = mmligner.calculate([mol1_protein, mol2_protein])
~/.conda/envs/pyscience/lib/python3.7/site-packages/opencadd/structure/superposition/engines/base.py in calculate(self, structures, *args, **kwargs)
29         """
30         assert len(structures) == 2
---> 31         return self._calculate(structures, *args, **kwargs)
     32 
     33     def _calculate(self, structures, *args, **kwargs):
~/.conda/envs/pyscience/lib/python3.7/site-packages/opencadd/structure/superposition/engines/mmligner.py in _calculate(self,
structures, *args, **kwargs)
     94             )
     95             # We need access to the temporary files at parse time!
---> 96             result = self._parse_metadata(output.decode())
     97             superposed_models = self._calculate_transformed(structures, result["metadata"])
     98             result["superposed"] = superposed_models
~/.conda/envs/pyscience/lib/python3.7/site-packages/opencadd/structure/superposition/engines/mmligner.py in
_parse_metadata(self, output)
    142         # if the patched mmligner is used (check /devtools/conda-recipes/mmligner)
    143         # -- this will fail in CI for now --
--> 144         translation = fixed_com - moving_com
    145 
    146         alignment = fasta.FastaFile()
UnboundLocalError: local variable 'fixed_com' referenced before assignment

Is there a problem with the installation?

**Comment by jaimergp** [comment]
Nop, probably a bug in our wrapper. I'll create an issue to deal with this in a separate PR. Thanks for the report!

**Pull request merged**